### PR TITLE
userv: Add integrated methods for QAT

### LINKED ISSUE #108: Add integrated methods for QAT
Determine whether to enable zip function and zip tool (cpu or qat) by reading environment variables.

### PR DESCRIPTION
**Pull request by Yanbo0101**
This PR is to fix #108 , add a method for determining whether to enable zip function and zip tools by reading environment variables.

### CODE REVIEW

File: cnap/core/cpu.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+"""A CPU module.
+
+This module provides a class to compress/decompress of frames.
```
Comment (Hulongyin) [inline cnap/core/cpu.py]: It might make more sense to make the module name related to compression

File: cnap/core/cpu.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    def __del__(self):
+        """Destroy a ZipBase object."""
+
+    @abstractmethod
+    def compress(self, src: bytes) -> bytes:
+        """Compress.
+
+        The method is to compress frame.
+
```
Comment (Hulongyin) [inline cnap/core/cpu.py]: Add `Args` and `Returns` in comment

File: cnap/core/cpu.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            NotImplementedError: If the subclasses don't implement the method.
+        """
+        raise NotImplementedError("Subclasses should implement connect() method.")
+
+    @abstractmethod
+    def decompress(self, src: bytes) -> bytes:
+        """Decompress.
+
+        The method is to decompress frame.
+
```
Comment (Hulongyin) [inline cnap/core/cpu.py]: Same

File: cnap/core/inferqueue.py [hunk 4]
```diff
@@ -376,7 +362,7 @@ def publish_frame(self, topic: str, frame: Frame) -> None:
             raise RuntimeError(e) from e
```
Comment (Hulongyin) [inline cnap/core/inferqueue.py]: Also remove `to_blob` here.

File: cnap/userv/inference.py [hunk 5]
```diff
@@ -263,6 +267,35 @@ def pipeline_manager(self) -> PipelineManager:
...
+        
+        Returns:
+            ZipBase: The zip tool instance.
+
+        Raises:
+            NotImplementedError: If the runtime database type is not supported.
+            qat.exceptions.RuntimeError: If QAT initialize fails.
+        """
+        if self._zip_tool is None:
+            zip_info = self.get_env("ZIP_INFO","false")
```
Comment (Hulongyin) [inline cnap/userv/inference.py]: `ZIP_ENABLE` or some other name may make more sense

File: cnap/userv/inference.py [hunk 6]
```diff
@@ -263,6 +267,35 @@ def pipeline_manager(self) -> PipelineManager:
...
+            if zip_info == "true":
+                zip_tool = self.get_env("ZIP_TOOL","CPU")
+                if zip_tool == "QAT":
+                    self._zip_tool  = QATZip()
```
Comment (Hulongyin) [inline cnap/userv/inference.py:289]: The `QATZip` Class should also be a subclass of the `ZipBase` class

File: cnap/userv/streaming.py [hunk 7]
```diff
@@ -210,6 +215,35 @@ def infer_engine_info(self) -> Optional[InferenceInfo]:
...
+        
+        Returns:
+            ZipBase: The zip tool instance.
+
+        Raises:
+            NotImplementedError: If the runtime database type is not supported.
+            qat.exceptions.RuntimeError: If QAT initialize fails.
+        """
+        if self._zip_tool is None:
+            zip_info = self.get_env("ZIP_INFO","false")
```
Comment (Hulongyin) [inline cnap/userv/streaming.py]: `ZIP_ENABLE` or some other name may make more sense

File: cnap/userv/streaming.py [hunk 8]
```diff
@@ -210,6 +215,35 @@ def infer_engine_info(self) -> Optional[InferenceInfo]:
...
+            if zip_info == "true":
+                zip_tool = self.get_env("ZIP_TOOL","CPU")
+                if zip_tool == "QAT":
+                    self._zip_tool  = QATZip()
```
Comment (Hulongyin) [inline cnap/userv/streaming.py:237]: The `QATZip` Class should also be a subclass of the ZipBase class

File: cnap/userv/inference.py [hunk 9]
```diff
@@ -423,18 +461,45 @@ def reset_frame_counts(self) -> None:
         self.infer_frame_count_sum = 0
         self.drop_frame_count_sum = 0
 
+    def process_frame(self, frame: bytes) -> Optional[Frame]:
```
Comment (Hulongyin) [inline cnap/userv/inference.py]: return `Frame`, Optional not needed

**Comment by leyao-daily** [comment]
what in my consideration is that, we may have many potential crypto devices in the future, maybe we create a crypto base class to handle all these (cpu based, qat based...) instead of creating so many qat/cpu files in core, we can create a crypto folder under core and move all related files in to that folder to make it clear. what do you think?

**Comment by Yanbo0101** [comment]
> what in my consideration is that, we may have many potential crypto devices in the future, maybe we create a crypto base class to handle all these (cpu based, qat based...) instead of creating so many qat/cpu files in core, we can create a crypto folder under core and move all related files in to that folder to make it clear. what do you think?

Sure, I will do this, and it might be more appropriate to call it a zip folder?

**Comment by leyao-daily** [comment]
> > what in my consideration is that, we may have many potential crypto devices in the future, maybe we create a crypto base class to handle all these (cpu based, qat based...) instead of creating so many qat/cpu files in core, we can create a crypto folder under core and move all related files in to that folder to make it clear. what do you think?
> 
> Sure, I will do this, and it might be more appropriate to call it a zip folder?

core/crypto/(qat.py, cpu.py...?) or core/crypto/(zip, cipher, ...)

**Comment by Yanbo0101** [comment]
> > > what in my consideration is that, we may have many potential crypto devices in the future, maybe we create a crypto base class to handle all these (cpu based, qat based...) instead of creating so many qat/cpu files in core, we can create a crypto folder under core and move all related files in to that folder to make it clear. what do you think?
> > 
> > 
> > Sure, I will do this, and it might be more appropriate to call it a zip folder?
> 
> core/crypto/(qat.py, cpu.py...?) or core/crypto/(zip, cipher, ...)

how about core/accelerator/qat.py zip.py or core/compression/qat.py zip.py? the current feature is only related to compression and decompression.

File: cnap/userv/inference.py (hunk 9 above)
Comment (Yanbo0101) [reply cnap/userv/inference.py]: Fixed all. Thanks

**Comment by Yanbo0101** [comment]
> what in my consideration is that, we may have many potential crypto devices in the future, maybe we create a crypto base class to handle all these (cpu based, qat based...) instead of creating so many qat/cpu files in core, we can create a crypto folder under core and move all related files in to that folder to make it clear. what do you think?

Fixed. Thanks!

**Review by Hulongyin — APPROVED** [review]
LGTM. Thanks

**Pull request merged**